# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build ffmpeg-build gst-plugins-base-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15; lt-2.4.6)
libtoolize -c -f; aclocal -I m4 -I common/m4; autoheader; autoconf; automake -a -c
rm -rf gst-libs/ext
PKG_CONFIG_PATH=/dfb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/dfb/lib ./configure --with-system-libav --prefix=/dfb

# build
make

# install
install -d $DESTDIR/dfb/lib/gstreamer-1.0
install ext/libav/.libs/libgstlibav.so $DESTDIR/dfb/lib/gstreamer-1.0

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/gst-libav-build.pc << EOF
Name: gst-libav
Version: 1.8.3
Description: FFmpeg GStreamer plugin
Requires: $REQUIRES

exec=\\
/dfb/lib/gstreamer-1.0/libgstlibav.so
EOF

exit

# patch
--- gst-libav-1.8.3.orig/configure.ac
+++ gst-libav-1.8.3/configure.ac
@@ -25,7 +25,7 @@
 AC_CONFIG_HEADERS([config.h])
 
 dnl AM_MAINTAINER_MODE only provides the option to configure to enable it
-AM_MAINTAINER_MODE([enable])
+AM_MAINTAINER_MODE
 
 dnl sets host_* variables
 AC_CANONICAL_HOST
@@ -485,7 +485,6 @@
 ext/libav/Makefile
 ext/libswscale/Makefile
 gst-libs/Makefile
-gst-libs/ext/Makefile
 docs/Makefile
 docs/plugins/Makefile
 docs/version.entities
@@ -495,6 +494,9 @@
 pkgconfig/Makefile
 pkgconfig/gstreamer-plugins-libav-uninstalled.pc
 )
+if test "$with_system_libav" != "yes"; then
+  AC_CONFIG_FILES(gst-libs/ext/Makefile)
+fi
 AC_OUTPUT
 
 ORC_OUTPUT

# source
https://gstreamer.freedesktop.org/src/gst-libav/gst-libav-1.8.3.tar.xz
